torchaudio/functional/filtering.py

```diff
@@ -1478,6 +1482,16 @@ def vad(
 
     # pack batch
     shape = waveform.size()
+
+    if len(shape) > 2:
+        warnings.warn(
+            "Expected input tensor shape length of 1 for single channel"
+            f" or 2 for multi-channel. Got {len(shape)} instead.\n"
+            "Batch semantics is not supported yet.\n"
```

Comment:
* Here, I think `dimension` (as in `ndim` attribute) is more intuitive than `shape length`.
* > Batch semantics is not supported yet.

    Let's not say `"yet"`, as we do not know if we can support with the current interface.
* In general, we do not use `\n` in warnings. But if you find counter example, that's fine as well.